ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/AbstractRecord.java

```diff
@@ -473,9 +473,21 @@ public static AbstractRecord create (int type)
         }
 	    catch (final Throwable ex)
 	    {
-	        ex.printStackTrace();
-	        
-	        return null;
+			try {
+				Class rt = RecordType.typeToClass(type);
+				String className = (rt == null) ? "null" : rt.getName();
+				int mb = 1024 * 1024;
+				Runtime runtime = Runtime.getRuntime();
+				long used = (runtime.totalMemory() - runtime.freeMemory()) / mb;
+				long free = runtime.freeMemory() / mb;
+
+				tsLogger.i18NLogger.warn_coordinator_AbstractRecord_create_failure(
+						type, className, used, free, ex);
```

Comment:
I think this should log ex.printStackTrace();after the warn as before in case people are watching system.err for the stack trace...

Reply:
Try that last commit instead

Reply:
My opinion is that we should try to avoid use of the `Exception.printStackTrace()` calls when logging is available. It's usual good practice to java to use loggers
https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#avoidprintstacktrace
Logger offers better control on when and what will be shown, logger is usually used to gather and redirect logs over network etc.

Reply:
On the one hand I do agree that we should definitely make sure any new log statements follow best practice, once the code is in the wild we have to consider whether making the change outweighs the benefit of users who may be relying on the old behaviour. This is not a precise decision always as the pro/con have to be evaluated each time.